src/server/AppManager.ts

```diff
@@ -1032,6 +1041,30 @@ export class AppManager {
 
         return result;
     }
+
+    private checksum(str: string, alg = 'SHA256'): string {
+        return createHash(alg).update(str).digest('hex');
+    }
+
+    private stringifyOrdered(obj: any): string {
+        return JSON.stringify(obj, Object.keys(obj).sort());
```

Comment:
On the app update flow, this will cause the descriptor string to include the signature itself to be hashed. I'm wondering if this won't cause problems since we'll be updating that signature after calculating the hash.

Reply:
Maybe this would be enough?

```suggestion
        return JSON.stringify(obj, Object.keys(obj).filter(key => key !== 'signature').sort());
```